am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java

```diff
@@ -118,6 +118,45 @@ void whenRevokingResourceAccessThatDoesNotExistNoErrorExpected() {
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
     }
 
+    @Test
+    void whenRevokingResourceWithNonExistentRelationshipShouldNotRemoveAnyRecordFromDatabase() {
+        grantExplicitResourceAccess(resourceId, "");
+
+        service.revokeResourceAccess(ExplicitAccessMetadata.builder()
+            .resourceId(resourceId)
+            .accessorId(accessorId)
+            .accessorType(USER)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attribute(JsonPointer.valueOf(""))
+            .securityClassification(PUBLIC)
+            .relationship("NonExistentRelationship")
+            .build());
+
+        assertThat(databaseHelper.findExplicitPermissions(resourceId)).hasSize(1)
+            .extracting(ExplicitAccessRecord::getRelationship).contains(ROLE_NAME);
```

Comment:
I'd rather want to have assertion around resource ID that is unique.

Reply:
👍